R/Phenotyping.R

```diff
@@ -348,4 +351,84 @@ ExpandGeneList <- function(genes, verbose = TRUE) {
 	}
 
 	return(ret)
+}
+
+#' @title .FindGeneSetsContaining
+#'
+#' @description A helper function that Takes an input gene and identifies any entries matching registered gene sets. This function is the complement to GetGeneSet.
+#' @param genes A vector of genes 
+.FindGeneSetsContaining <- function(genes){
+  
+  matchingGeneSets <- unlist(RIRA:::pkg.env$GENE_SETS)[ unlist(RIRA:::pkg.env$GENE_SETS) %in% genes]
+  gene_matches_df <- data.frame(gene = matchingGeneSets, set = names(matchingGeneSets))
+  #strip the trailing numbers from geneset names (and occasional peroids at the end of gene set signatures)
+  gene_matches_df$set <- gsub("\\.?[0-9]*$", "", gene_matches_df$set) 
+  
+  gene_matches_df <- gene_matches_df |> 
+    dplyr::group_by(gene) |>
+    dplyr::mutate(geneset_union = paste(set, collapse = ", "))
+  unique_gene_set_pairs <- unique.data.frame(gene_matches_df[,c("gene", "geneset_union")])
+  
+  return(unique_gene_set_pairs)
+}
+
+#' @title MakePhenotypingDotPlot
+#'
+#' @description Creates a DotPlot using custom gene sets and attempts to coarsely group gene sets by cell type. 
+#' @param yField The grouping variable used to calculate the average expression of genes and the y axis of the DotPlot.
+#' @param scaled A boolean defining whether to color dots by scaled expression or unscaled expression.
+#' @param gene_lists A vector of gene lists (defined by .RegisterGeneSet) to be queried and their genes be plotted.
+MakePhenotypingDotPlot <- function(seuratObj,
+                                   yField = 'ClusterNames_0.2',
+                                   scaled = T, 
+                                   gene_lists = c('Cytotoxicity', 'EffectorCytokines')
```

Comment:
@bbimber, I added support for supplying a vector of GeneSets. Do you have any gene sets that you'd want this to include by default? 

The warning from GetGeneSet provides some safety on nonsense input gene names.